test/powershell/engine/Remoting/RoleCapabilityFiles.Tests.ps1

```diff
@@ -0,0 +1,118 @@
+##
+## PowerShell Remoting Endpoint Role Capability Files Tests
+##
+
+Describe "Remote session configuration RoleDefintion RoleCapabilityFiles key tests" -Tags "Feature" {
+
+    BeforeAll {
+
+        if (!$IsWindows)
+        {
+            $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+            $PSDefaultParameterValues["it:skip"] = $true
+        }
+        else
+        {
+            [string] $RoleCapDirectory = (New-Item -Path "$TestDrive\RoleCapability" -ItemType Directory -Force).FullName
+
+            [string] $GoodRoleCapFile = "$RoleCapDirectory\TestGoodRoleCap.psrc"
+            New-PSRoleCapabilityFile -Path $GoodRoleCapFile -VisibleCmdlets 'Get-Command','Get-Process','Clear-Host','Out-Default','Select-Object','Get-FormatData','Get-Help'
+
+            [string] $BadRoleCapFile = "$RoleCapDirectory\TestBadRoleCap.psrc"
+            New-PSRoleCapabilityFile -Path $BadRoleCapFile -VisibleCmdlets *
+            [string] $BadRoleCapFile = $BadRoleCapFile.Replace('.psrc', 'psbad')
+
+            [string] $PSSessionConfigFile = "$RoleCapDirectory\TestConfig.pssc"
+        }
+    }
+
+    AfterAll {
+
+        if (!$IsWindows)
+        {
+            $global:PSDefaultParameterValues = $originalDefaultParameterValues
+        }
+    }
+
+    It "Verifies missing role capability file error" {
+
+        New-PSSessionConfigurationFile -Path $PSSessionConfigFile -RoleDefinitions @{
+            Administrators = @{ RoleCapabilityFiles = "$RoleCapDirectory\NoFile.psrc" }
+        }
+
+        try
+        {
+            $iss = [initialsessionstate]::CreateFromSessionConfigurationFile($PSSessionConfigFile, { $true })
+            throw 'No Exception!'
+        }
+        catch
+        {
+            [System.Management.Automation.MethodInvocationException] $expectedException = [System.Management.Automation.MethodInvocationException] $_.Exception
+            if ($expectedException -ne $null)
+            {
+                ([System.Management.Automation.PSInvalidOperationException] $expectedException.InnerException).ErrorRecord.FullyQualifiedErrorId | Should Be 'CouldNotFindRoleCapabilityFile'
+            }
+            else
+            {
+                throw 'Unexpected Exception'
```

Comment:
It is not sufficient because L53 is invoked only if the correct exception is found.  If it is not the expected exception then the test must fail.

I have to say that not filtering on the desired exception feels like an arbitrary and unneeded change.  It did exactly what I intended and with less code.